### PR TITLE
Bring back `max-w-screen-*` utilities

### PR DESCRIPTION
**Pull request by philipp-spiess**
This PR brings back the `max-w-screen-*` utilities from v3 that read from the `--breakpoint` namespace. 

Since these utilities are only added back for compatibility reasons, it's put into the compatibility layer.

Note that this does not do Intellisense recommendations for the functional utility. 

## Test Plan

Unit tests are upgraded including some from the compat test that extends the `--breakpoint` namespace from `screen` keys. Also tested this in the Vite playground.

### CODE REVIEW

File: packages/tailwindcss/src/compat/legacy-utilities.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    ['b', 'bottom'],
+    ['bl', 'bottom left'],
+    ['l', 'left'],
+    ['tl', 'top left'],
+  ]) {
+    designSystem.utilities.static(`bg-gradient-to-${value}`, () => [
+      decl('--tw-gradient-position', `to ${direction} in oklch,`),
+      decl('background-image', `linear-gradient(var(--tw-gradient-stops))`),
+    ])
+  }
```
Comment (thecrypticace) [inline packages/tailwindcss/src/compat/legacy-utilities.ts:19]: These need to be removed from the utilities.ts file

**Review by thecrypticace — CHANGES_REQUESTED** [review]
left a note

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Bring back support for color opacity modifiers to read from `--opacity-*` theme values ([#14278](https://github.com/tailwindlabs/tailwindcss/pull/14278))
 
+### Added
+
+- Reintroduce `max-w-screen-*` utilities that read from the `--breakpoint` namespace ([#15013](https://github.com/tailwindlabs/tailwindcss/pull/15013))
```
Comment (adamwathan) [inline CHANGELOG.md:16]:
```suggestion
- Reintroduce `max-w-screen-*` utilities that read from the `--breakpoint` namespace as deprecated utilities ([#15013](https://github.com/tailwindlabs/tailwindcss/pull/15013))
```

File: packages/tailwindcss/src/compat/legacy-utilities.ts (hunk 1 above)
Comment (adamwathan) [reply packages/tailwindcss/src/compat/legacy-utilities.ts:19]: Good catch, removed!

**Review by adamwathan — APPROVED** [review]

**Pull request merged**